linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -21,245 +21,254 @@
+        - name: "Add autoinstall to BIOS boot kernel command for Ubuntu live server"
           ansible.builtin.replace:
-            path: "{{ unattend_iso_cache }}/grub.cfg"
-            regexp: '(.*vmlinuz)(.*)'
+            path: "{{ unattend_iso_cache }}/txt.cfg"
+            regexp: '(.*initrd)(.*)'
             replace: "\\1 autoinstall \\2"
 
-        - name: "Set timeout to 5 seconds at boot menu"
-          ansible.builtin.replace:
-            path: "{{ unattend_iso_cache }}/grub.cfg"
-            regexp: 'set timeout=.*'
-            replace: "set timeout=5"
-
-        - name: "Update md5sum for UEFI boot config file"
+        - name: "Update md5sum for BIOS boot config file"
           ansible.builtin.shell: |
-            md5=`md5sum grub.cfg | awk '{print $1}'`
-            sed -i "/.\/boot\/grub\/grub.cfg/ s/^[^ ]*/$md5/" md5sum.txt
+            md5=`md5sum txt.cfg | awk '{print $1}'`
+            sed -i "/.\/isolinux\/txt.cfg/ s/^[^ ]*/$md5/" md5sum.txt
           args:
             chdir: "{{ unattend_iso_cache }}"
-
-        - name: "Update BIOS boot config file"
-          block:
-            - name: "Add autoinstall to BIOS boot kernel command for Ubuntu live server"
-              ansible.builtin.replace:
-                path: "{{ unattend_iso_cache }}/txt.cfg"
-                regexp: '(.*initrd)(.*)'
-                replace: "\\1 autoinstall \\2"
-
-            - name: "Update md5sum for BIOS boot config file"
-              ansible.builtin.shell: |
-                md5=`md5sum txt.cfg | awk '{print $1}'`
-                sed -i "/.\/isolinux\/txt.cfg/ s/^[^ ]*/$md5/" md5sum.txt
-              args:
-                chdir: "{{ unattend_iso_cache }}"
-          when: "'/isolinux/txt.cfg' in iso_files_to_update"
-
-      when: unattend_install_conf is match('Ubuntu/Server')
-
-    - name: "Rebuild ISO image with unattend install config file"
-      ansible.builtin.shell: "{{ rebuild_iso_cmd }}"
-      when:
-        - rebuild_iso_cmd is defined
-        - rebuild_iso_cmd
-
-    # Unmount the original ISO
-    - include_tasks: ../../common/local_unmount.yml
-      vars:
-        mount_path: "{{ tmp_iso_mount_dir }}"
-
-    - name: "Cleanup original ISO files and copied files"
-      ansible.builtin.file:
-        path: "{{ item }}"
-        state: absent
-      with_items:
-        - "{{ src_iso_file_path }}"
-        - "{{ tmp_iso_mount_dir }}"
-  become: true
+      when: ubuntu_bios_cfg_exist
+    
+    - name: Customize the ISO
+      community.general.iso_customize:
+        src_iso: "{{ src_iso_file_path }}"
+        dest_iso: "{{ rebuilt_unattend_iso_path }}"
+        add_files:
+          - src_file: "{{ src_iso_file_dir }}/grub.cfg"
+            dest_file: "boot/grub/grub.cfg"
+          - src_file: "{{ src_iso_file_dir }}/md5sum.txt"
+            dest_file: "md5sum.txt"
+          - src_file: "{{ src_iso_file_dir }}/txt.cfg"
+            dest_file: "isolinux/txt.cfg"
+      when: ubuntu_bios_cfg_exist
```

Comment:
This task can be added in above block as they have same when condition

Reply:
Thanks. Done